### PR TITLE
Expand a bit the documentation about transformers.

### PR DESCRIPTION
**Pull request by Carreau**
Mostly answering questions I had myself while trying to write some stuff
and while trying to get a simple example working.

### CODE REVIEW

File: IPython/core/inputsplitter.py [hunk 1]
```diff
@@ -687,7 +686,6 @@ def push(self, lines):
         """
 
         # We must ensure all input is pure unicode
-        lines = cast_unicode(lines, self.encoding)
```
Comment (takluyver) [inline IPython/core/inputsplitter.py]: Why is this changed in a doc PR? If you're sure removing it is OK, please remove the comment too.

File: docs/source/config/inputtransforms.rst [hunk 2]
```diff
@@ -71,6 +71,32 @@ The decorator returns a factory function which will produce instances of
...
 function.
 
+Transforming a full block
+-------------------------
+
+Transforming a full block of python code is possible by implementing a
+:class:`~IPython.core.inputtransformer.Inputtransformer` and overwriting the
+``push`` and ``reset`` methods. The reset method should send the full block of
+transformed text. As an example a transformer the reversed the lines from last
+to first.
```
Comment (takluyver) [inline docs/source/config/inputtransforms.rst]: We should say that this will break the automatic 'is this input complete' detection in console interfaces - with a transformer like this added, it will always think more input is needed, and you will need to use a shortcut to force it to execute.

File: docs/source/config/inputtransforms.rst [hunk 3]
```diff
@@ -79,9 +105,28 @@ sent each line in turn, followed by ``None`` to reset it. It can yield lines, or
...
+
+    ip = get_ipython()
+
+    ip.input_splitter.logical_line_transforms.append(noop())
+    ip.input_transformer_manager.logical_line_transforms.append(noop())
+
 This code in IPython strips a constant amount of leading indentation from each
 line in a cell::
 
+    from IPython.core.inputtransformer import CoroutineInputTransformer
```
Comment (takluyver) [inline docs/source/config/inputtransforms.rst]: Do you want to add the import to your new example above too?

File: docs/source/config/inputtransforms.rst [hunk 4]
```diff
@@ -79,9 +105,28 @@ sent each line in turn, followed by ``None`` to reset it. It can yield lines, or
 ``None`` if it is accumulating text to yield at a later point. When reset, it
 should give up any code it has accumulated.
 
+You may use :meth:`CoroutineInputTransformer.wrap` to simplify the creation of
+such a transformer.
+
+Here is a simple :class:`CoroutineInputTransformer` that can be though of be
```
Comment (willingc) [inline docs/source/config/inputtransforms.rst]: s/though/thought

Comment (willingc) [inline docs/source/config/inputtransforms.rst]: ~~be~~

File: docs/source/config/inputtransforms.rst [hunk 5]
```diff
@@ -71,6 +71,39 @@ The decorator returns a factory function which will produce instances of
 :class:`~IPython.core.inputtransformer.StatelessInputTransformer` using your
 function.
 
+Transforming a full block
+-------------------------
+
+.. warning::
+
+    Transforming a full block at once will break the automatic detection of
+    wether a block of code is complete in interface relying on this
```
Comment (takluyver) [inline docs/source/config/inputtransforms.rst]: - whether
- interfaces

File: docs/source/config/inputtransforms.rst [hunk 6]
```diff
@@ -71,6 +71,39 @@ The decorator returns a factory function which will produce instances of
...
+    functionality, as for example terminal IPython. You will need to use a
```
Comment (takluyver) [inline docs/source/config/inputtransforms.rst]: 'as for example' doesn't work in English. You could use 'such as', 'including', or 'for example'.

('such as, for example, foo' is valid, but it's too verbose to read nicely here, IMO)

Comment (Carreau) [inline docs/source/config/inputtransforms.rst]: done,

File: docs/source/config/inputtransforms.rst (hunk 5 above)
Comment (Carreau) [inline docs/source/config/inputtransforms.rst]: done.
